Bonsai.Shaders/BindTexture.cs

```diff
@@ -58,35 +58,40 @@ public bool IndexSpecified
             get { return Index.HasValue; }
         }
 
-        IObservable<TSource> Process<TSource>(IObservable<TSource> source, Action<TSource> update)
+        IObservable<TSource> Process<TSource>(IObservable<TSource> source, bool allowUnbind)
         {
             return Observable.Defer(() =>
             {
-                var texture = default(Texture);
-                var textureName = default(string);
+                var cachedTexture = default(Texture);
+                var cachedTextureName = default(string);
                 return source.CombineEither(
                     ShaderManager.ReserveShader(ShaderName),
                     (input, shader) =>
                     {
-                        if (textureName != TextureName)
+                        if (cachedTextureName != TextureName)
                         {
-                            textureName = TextureName;
-                            texture = !string.IsNullOrEmpty(textureName)
-                                ? shader.Window.ResourceManager.Load<Texture>(textureName)
+                            cachedTextureName = TextureName;
+                            cachedTexture = !string.IsNullOrEmpty(cachedTextureName)
+                                ? shader.Window.ResourceManager.Load<Texture>(cachedTextureName)
                                 : null;
                         }
 
-                        if (texture != null)
+                        var texture = cachedTexture ?? input as Texture;
+
+                        if (texture != null || allowUnbind)
```

Comment:
We actually don't want the unbind behavior for this patch, as that would be a breaking change. I agree the nitpick about the performance of the `as` operator is most likely irrelevant, so probably the following would be enough in a single method:

```c#
var activeTexture = texture ?? input as Texture;
if (activeTexture != null)
{
    var textureId = Index is int index
        ? ((TextureSequence)activeTexture).Textures[index]
        : activeTexture.Id;
    shader.Update(() =>
    {
        GL.ActiveTexture(TextureSlot);
        GL.BindTexture(TextureTarget, textureId);
    });
}
```

I opted to create a new local rather than rename the cache variable to minimize diffs, but that is just a style preference.

Reply:
The current implementation supports this unbind behavior:

https://github.com/bonsai-rx/bonsai/blob/ced27db65e266c0eab41a41f55cb22b70ce553b6/Bonsai.Shaders/BindTexture.cs#L139

Also IMO it's better for the "special" variable to have the unique name. It'd be easy to overlook someone accidentally using `texture` instead of `activeTexture` down below. It also makes it more clear what that closed variable is actually for and avoids mistakes like the one I made earlier with caching the `input`.

I definitely get wanting to avoid diff churn, but things should be optimized for people reading the current code not its history.

Reply:
> The current implementation supports this unbind behavior

Ah, you are completely correct! In that case let's leave it then.

Completely agree on optimizing for reading, my main reason for trying to minimize diffs is making it easier to use git blame tools. Anyway, in this case not critically important.